src/test/java/io/stargate/sgv2/jsonapi/service/embedding/operation/DataVectorizerTest.java

```diff
@@ -134,6 +139,87 @@ public void testWithBothVectorFieldValues() {
         throw new RuntimeException(e);
       }
     }
+
+    @Test
+    public void testWithUnmatchedVectorsNumber() {
+      TestEmbeddingProvider testProvider =
+          new TestEmbeddingProvider() {
+            @Override
+            public Uni<List<float[]>> vectorize(
+                List<String> texts,
+                Optional<String> apiKey,
+                EmbeddingRequestType embeddingRequestType) {
+              List<float[]> customResponse = new ArrayList<>();
+              texts.forEach(t -> customResponse.add(new float[] {0.5f, 0.5f, 0.5f}));
+              // add additional vector
+              customResponse.add(new float[] {0.5f, 0.5f, 0.5f});
+              return Uni.createFrom().item(customResponse);
+            }
+          };
+      List<JsonNode> documents = new ArrayList<>();
+      for (int i = 0; i < 2; i++) {
+        documents.add(objectMapper.createObjectNode().put("$vectorize", "test data"));
+      }
+      DataVectorizer dataVectorizer =
+          new DataVectorizer(
+              testProvider, objectMapper.getNodeFactory(), Optional.empty(), collectionSettings);
+      try {
+        Throwable failure =
+            dataVectorizer
+                .vectorize(documents)
+                .subscribe()
+                .withSubscriber(UniAssertSubscriber.create())
+                .awaitFailure()
+                .getFailure();
+        assertThat(failure)
+            .isInstanceOf(JsonApiException.class)
+            .hasFieldOrPropertyWithValue("errorCode", ErrorCode.EMBEDDING_PROVIDER_INVALID_RESPONSE)
+            .hasFieldOrPropertyWithValue(
+                "message",
+                "The configured Embedding Provider for this collection return an invalid response: Embedding provider 'custom' didn't return the expected number of embeddings. Expect: '2'. Actual: '3'");
+      } catch (Exception e) {
+        throw new RuntimeException(e);
```

Comment:
Could instead make test method `throws Exception` and leave out try-catch block.